src/table/utils/__tests__/copy-utils.spec.ts

```diff
@@ -15,7 +15,21 @@ describe('copyCellValue: ', () => {
   it('should not send console error when copying cell values', () => {
     evt = {
       target: {
-        children: [],
+        parentElement: {
+          firstChild: {
+            textContent: 'Test',
+          },
+        },
+        children: [
+          {
+            firstChild: {
+              textContent: 'Test',
+            },
+          },
+        ],
+        classList: {
+          contains: () => true,
+        },
```

Comment:
I spent some times on adding more testing scenarios here but didn't go well. The way clipboard is tested in jest is very limited and only through mocks. clipboard doesn't really copy any value if the context is not secure. I couldn't find a way to force it to do. I think integration tests are more suitable for copy to clipboard cases

Reply:
you can use `.className` instead of `.classList` and you get a string like this `'class1 class2 class3...'` and then you can use the string method `includes` instead

Reply:
... well, replacing it with className doesn't make a big change. The issue is clipboard itself, it is only available in secure context like https or localhost so in a unit test even without mocking it will not do copying so I cannot verify the string is copied ( to test that the cell value is copied as expected ).

Reply:
ah ok, I misread the comment. Something like this works:
```
  let writeMock: (arg: string) => void;
  beforeEach(() => {
    console.log = jest.fn();
    writeMock = jest.fn();
    Object.assign(navigator, {
      clipboard: {
        writeText: writeMock,
      },
    });
  });
```
Also, you have one of the `beforeEach` should be a `afterEach`

Reply:
no problem, yes we already have the similar mock, I actually meant that I would like to add more tests on what the clipboard result could be. It's only because we have few conditions to catch the right cell value so wanted to verify that the copied values are the expected ones, something like writeText and then later check the result by readText(), but because of clipboard limitations it is not really possible.

Reply:
I mean whatever happens after we call writeText shoulldn't really be tested here. We want to make sure that writeText is called with the correct value. That is tested yet. Also we should test all three cases of setting the value.
We could possibly do a test in sense-client later for this, but not sure we should do integration tests on all the cases here